jwst/barshadow/bar_shadow.py

```diff
@@ -11,94 +11,109 @@
 
 SLITRATIO = 1.15     # Ratio of slit spacing to slit height
 
+
 def do_correction(input_model, barshadow_model):
     """Do the Bar Shadow Correction
 
     Parameters
     ----------
-    input_model: MultiSlitModel datamodel object
+    input_model : `~jwst.datamodels.MultiSlitModel`
         science data model to be corrected
 
-    barshadow_model: BarshadowModel datamodel object
-        bar shadow datamodel from reference file
+    barshadow_model : `~jwst.datamodels.BarshadowModel`
+        bar shadow data model from reference file
 
     Returns
     -------
-    output_model: MultiSlitModel datamodel object
-        Science datamodel with bar shadow extensions added
-
+    output_model : `~jwst.datamodels.MultiSlitModel`
+        Science data model with correction applied and barshadow extensions added
     """
-#
-# Input is a MultiSlitModel science data model 
-# A MultislitModel has a member .slits that behaves like
-# a list of Slits, each of which has several data arrays and
-# keyword metadata items associated 
-#
-# At this point we're going to have to assume that a Slit is composed
-# of a set of slit[].nshutters in a vertical line
-#
-# Reference file information is a 1x1 ref file and a 1x3 ref file
-# Both the 1x1 and 1x3 ref files are 1001 pixel high and go from
-# -1 to 1 in their Y value WCS.  
-# 
-# 
-    exp_type = input_model.meta.exposure.type
-    log.info(exp_type)
+
+    # Input is a MultiSlitModel science data model.
+    # A MultislitModel has a member ".slits" that behaves like
+    # a list of Slits, each of which has several data arrays and
+    # keyword metadata items associated.
+    #
+    # At this point we're going to have to assume that a Slit is composed
+    # of a set of slit[].nshutters in a vertical line.
     #
+    # Reference file information is a 1x1 ref file and a 1x3 ref file.
+    # Both the 1x1 and 1x3 ref files are 1001 pixels high and go from
+    # -1 to 1 in their Y value WCS.
+
+    exp_type = input_model.meta.exposure.type
+    log.debug('EXP_TYPE = %s' % exp_type)
+
     # Create the pieces that are put together to make the barshadow model
     shutter_elements = create_shutter_elements(barshadow_model)
     w0 = barshadow_model.crval1
     wave_increment = barshadow_model.cdelt1
     y_increment = barshadow_model.cdelt2
     shutter_height = 1.0 / y_increment
-    # For each slitlet
+
+    # Loop over all the slits in the input model
```

Comment:
It looks like `input_model` gets modified in place in this `do_correction()` function.  And the caller from the step also doesn't make a copy beforehand.  Which means the data is getting modified in place?

It might make sense to have this function make a copy first then do the correction.

Reply:
Good point. Back when it was just attaching the correction array, having it modify the input in-place was not so bad, but now it should make a new copy, just like all other steps do,

Reply:
Latest commit now creates a copy.